### PR TITLE
test/bluefs: use close_writer to close the file handle

### PR DESCRIPTION
**Pull request by wonderpow**
Signed-off-by: wonderpow <wangbozhao@cmss.chinamobile.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by wonderpow** [comment]
retest this please.

File: src/test/objectstore/test_bluefs.cc [hunk 1]
```diff
@@ -106,7 +106,7 @@ TEST(BlueFS, write_read) {
     BlueFS::FileReaderBuffer buf(4096);
     ASSERT_EQ(9, fs.read(h, &buf, 0, 1024, &bl, NULL));
     ASSERT_EQ(0, strncmp("foobarbaz", bl.c_str(), 9));
-    delete h;
+    fs.close_writer(h);
```
Comment (tchaikov) [inline src/test/objectstore/test_bluefs.cc]: this does not compile
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/objectstore/test_bluefs.cc: In member function 'virtual void BlueFS_write_read_Test::TestBody()':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/objectstore/test_bluefs.cc:109:22: error: no matching function for call to 'BlueFS::close_writer(BlueFS::FileReader*&)'
     fs.close_writer(h);
                      ^
```

Comment (wonderpow) [reply src/test/objectstore/test_bluefs.cc]: thanks for your advice

File: src/os/bluestore/BlueFS.h [hunk 2]
```diff
@@ -459,6 +459,11 @@ class BlueFS {
     _close_writer(h);
   }
 
+  void close_reader(FileReader *h) {
+    std::lock_guard l(lock);
```
Comment (tchaikov) [inline src/os/bluestore/BlueFS.h]: why do we need to hold a lock for deleting `FileReader`?

Comment (wonderpow) [reply src/os/bluestore/BlueFS.h]: when we called the function such as "BlueFS::open_for_read", we need to hold a lock, so need to hold a lock for deleting FileReader.

Comment (tchaikov) [reply src/os/bluestore/BlueFS.h]: i am not convinced.

**Pull request closed without merging**